pkg/argocd/argocd.go

```diff
@@ -48,6 +55,35 @@ func AddClusterAddFlags(cmd *cobra.Command) (AddClusterCmd, error) {
 	return &addClusterImpl{root, args}, nil
 }
 
+var o = sync.Once{}
+
+func CheckAppSynced(ctx context.Context, f kube.Factory, ns, name string) (bool, error) {
+	rc, err := f.ToRESTConfig()
+	if err != nil {
+		return false, err
+	}
+
+	c, err := argocdcd.NewForConfig(rc)
+	if err != nil {
+		return false, err
+	}
+
+	app, err := c.ArgoprojV1alpha1().Applications(ns).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		se, ok := err.(*errors.StatusError)
+		if !ok || se.ErrStatus.Reason != metav1.StatusReasonNotFound {
+			return false, err
+		}
+
+		return false, nil
+	}
+
+	o.Do(func() {
```

Comment:
actually lets remove this message and just add a debug message with the status each iteration. Because this sync.Once complicates things (it won't work if you run it a second time for another resource for example).